calico_cni_test.go

```diff
@@ -200,6 +201,36 @@ var _ = Describe("CalicoCni", func() {
 					Expect(err).ShouldNot(HaveOccurred())
 				})
 			})
+			Context("when the ready flag is not set", func() {
+				It("should return error on Add", func() {
+					kv, err := calicoClient.Backend.Apply(
+						&model.KVPair{
+							Key:   model.ReadyFlagKey{},
+							Value: false,
+						})
+					Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("%v", kv))
+					_, session, _, _, _, _, err := CreateContainer(netconf, "", "")
+					Expect(err).Should(HaveOccurred())
+					Eventually(session).ShouldNot(gexec.Exit(0))
```

Comment:
This works, but it looks a bit weird using `ShouldNot` with `Eventually`, because `Eventually` only checks for the condition for a certain amount of time (I think it's 1 second by default) and if it's set to `ShouldNot` then it's possible the plugin exits with exit code 0 but we still don't catch it. I suggest asserting the expected error code (code `1`?) with `Eventually`